### PR TITLE
Release 4.0.0

### PR DESCRIPTION
**Pull request by mats-claassen**
Changes:

- MOB-686: Using SFSafariViewController to authenticate the user.
- MOB-624: User session cookie saved in SDK
- MOB-660: Improve error handling in SDK (#27, #29, #30)
- MOB-438: Support different domains for production and development (#25)
- MOB-588: Added default page when there is not internet connection (#24, #26, #28)
- MOB-439: Access token refreshing (#22)
- MOB-596: Create function with login type (sign up or sign in) (#21)
- MOB-573: Add back button to web view (#20)
- MOB-570: Open new tabs in Safari (#19)
- MOB-559: Make navigation bar cancel button optional (#17)
- MOB-533: Added API for adding a new ID (affiliation) (#15)
- MOB-534: Added API for adding a new Login (connection) (#14)
- MOB-442: Remove code that replaces null values with “Unknown” (#10)
- MOB-415: Add access to access token and save it in the keychain (#11)
- MOB-441: Change UIWebView for WKWebView (#12)

### CODE REVIEW

**Review by sdownie — APPROVED** [review]

**Review by sdownie — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -3,31 +3,93 @@ The ID.me WebVerify SDK for iOS is a library that allows you to verify a user's
...
+    let handled = IDmeWebVerify.sharedInstance().application(app, open: url, options: options)
+
+    if !handled {
+        // do something else
+    }
+    return handled
+}
+```
+
+If your app suuports a lower iOS version than 9.0 then you must also add these methods:
```
Comment (m-revetria) [inline README.md]: misspelling: *supports*

File: README.md [hunk 2]
```diff
@@ -3,31 +3,93 @@ The ID.me WebVerify SDK for iOS is a library that allows you to verify a user's
...
+    let handled = IDmeWebVerify.sharedInstance().application(app, open: url, options: options)
+
+    if !handled {
+        // do something else
+    }
+    return handled
+}
+
+func application(_ application: UIApplication, handleOpen url: URL) -> Bool {
+    let handled = IDmeWebVerify.sharedInstance().application(app, open: url, options: options)
```
Comment (m-revetria) [inline README.md]: `options` is not defined in this scope

File: README.md [hunk 3]
```diff
@@ -3,31 +3,93 @@ The ID.me WebVerify SDK for iOS is a library that allows you to verify a user's
...
+        // do something else
+    }
+    return handled
+}
+```
+
+If your app suuports a lower iOS version than 9.0 then you must also add these methods:
+```swift
+func application(_ application: UIApplication, open url: URL, sourceApplication: String?, annotation: Any) -> Bool {
+    let handled = IDmeWebVerify.sharedInstance().application(app, open: url, options: options)
```
Comment (m-revetria) [inline README.md]: `options` is not defined in this scope

File: README.md [hunk 4]
```diff
@@ -3,31 +3,93 @@ The ID.me WebVerify SDK for iOS is a library that allows you to verify a user's
...
 
-or download it on Github.
+or download it from Github and add it to your Xcode project.
 
 ## Setup
-1. Drag the **ID.me WebVerify SDK** folder into your project. This will add the following files:
-	- `IDmeWebVerify.h`
-	- `IDmeWebVerifyNavigationController.h`
-2. Import `IDmeWebVerify.h` into one of your projects, preferably an instance or subclassed instance of `UIViewController`.
+### Step 1
```
Comment (m-revetria) [inline README.md]: I would move this point to the installation section

File: README.md [hunk 5]
```diff
@@ -3,31 +3,93 @@ The ID.me WebVerify SDK for iOS is a library that allows you to verify a user's
...
+If you installed the SDK through Cocoapods you can skip this step. Drag the downloaded files to your Xcode project. If you are working with Swift you will have to import `IDmeWebVerify.h` in your ObjC Bridging Header. You will also have to somehow import [SAMKeychain](https://github.com/soffes/SAMKeychain) to your project as this SDK depends on it.
+
+### Step 2
+You must call `IDmeWebVerify.initialize(withClientID: String, clientSecret: String, redirectURI: String)` before using the SDK. You can do it in `application(didFinishLaunchingWithOptions:)` for example.
```
Comment (m-revetria) [inline README.md]: I would include the import statement as part of the example code

File: README.md [hunk 6]
```diff
@@ -3,31 +3,93 @@ The ID.me WebVerify SDK for iOS is a library that allows you to verify a user's
...
+
+## Examples
+To see a working example you can:
+- Download this repository
+- Open `WebVerifySample.xcodeproj`
+- Set your `clientID`, `clientSecret`, `redirectURI` and `scope` in `ViewController.m`
+- Replace `your_custom_scheme` with your `redirectURI` in `WebVerifySample-Info.plist` -> `URL Types` (or through *Project Navigator*)
 
 ## Execution
+Verification occurs through a SFSafariViewController which will open Safari for the user to log in. You must call `verifyUser(in: UIViewController, scope: String, withTokenResult: IDmeVerifyWebVerifyTokenResults)` to launch a SFSafariViewController for the user to authenticate. Take care not to call this method while another instance of that verification process is still under way as it will throw an exception.
```
Comment (m-revetria) [inline README.md]: Maybe you can link the RFC to give more context on why this was changed

File: README.md [hunk 7]
```diff
@@ -3,31 +3,93 @@ The ID.me WebVerify SDK for iOS is a library that allows you to verify a user's
...
-	- `IDmeWebVerifyNavigationController.h`
-2. Import `IDmeWebVerify.h` into one of your projects, preferably an instance or subclassed instance of `UIViewController`.
+### Step 1
+If you installed the SDK through Cocoapods you can skip this step. Drag the downloaded files to your Xcode project. If you are working with Swift you will have to import `IDmeWebVerify.h` in your ObjC Bridging Header. You will also have to somehow import [SAMKeychain](https://github.com/soffes/SAMKeychain) to your project as this SDK depends on it.
+
+### Step 2
+You must call `IDmeWebVerify.initialize(withClientID: String, clientSecret: String, redirectURI: String)` before using the SDK. You can do it in `application(didFinishLaunchingWithOptions:)` for example.
+
+### Step 3
+You must handle the redirect calls from the SDK. For this you have to add your `redirectURI` as handled URL scheme. Go to *Project Navigator* -> *Select your target* -> *Info* -> *URL Types* -> *New* and add your redirectURI's scheme in *URL Schemes*.
```
Comment (sdownie) [inline README.md]: @mats-claassen Typo "as handled URL scheme"

File: README.md [hunk 8]
```diff
@@ -3,31 +3,93 @@ The ID.me WebVerify SDK for iOS is a library that allows you to verify a user's
...
+For example: if your redirectURI is `my_custom_scheme://callback`, enter `my_custom_scheme` in *URL Schemes*.
```
Comment (sdownie) [inline README.md]: @mats-claassen  Rename "For example" to "Example"

File: README.md (hunk 5 above)
Comment (m-revetria) [reply README.md]: Maybe you can add an explanation on what are those parameters, https://github.com/IDme/ID.me-WebVerify-SDK-Android/pull/33/files#diff-04c6e90faac2675aa89e2176d2eec7d8R84

**Review by sdownie — CHANGES_REQUESTED** [review]
see comments

File: README.md [hunk 9]
```diff
@@ -3,31 +3,93 @@ The ID.me WebVerify SDK for iOS is a library that allows you to verify a user's
...
-- **SDK Version:** 3.2.0 (June 24, 2016)
+- **SDK Version:** 4.0.0 (March 28, 2017)
 - **Maintained By:** [ID.me](http://github.com/IDme)
 
 For more information please email us at mobile@id.me or visit us at http://developer.id.me.
 
-## Changelog (3.2.0)
-- Changed parameter from affiliationType to scope (NSString)
+## Changelog
+The changelog can be found in [CHANGELOG.md](https://github.com/IDme/ID.me-WebVerify-SDK-iOS/blob/master/CHANGELOG.md)
```
Comment (mirland) [inline README.md]: you shouldn't use the full path

**Review by sdownie — APPROVED** [review]

**Pull request merged**